kod/object/item/passitem/spelitem.kod

```diff
@@ -216,8 +216,16 @@ messages:
       oSpell = Send(SYS,@FindSpellByNum,#num=viSpellEffect);
 
       % Get proper targets according to spell, assume that apply_on is our basic target.
-      lTargets = Send(oSpell,@GetTargets,#who=what,#lTargets=[apply_on]);
 
+      if IsClass(self,&Scroll)
```

Comment:
Rather than a gross IsClass, this looks like a prime candidate for a new little function the Scroll can override.  GetTargetsForSpell or something like that.